app/controllers/groupings_controller.rb

```diff
@@ -5,24 +5,34 @@ def show
     @parent = @grouping.parent_grouping
     @grouping_name = @grouping.name
     @children = @grouping.children
+    @background = @grouping.background.url
   end
 
   def edit
     @grouping_name = @grouping.name
     @children = @grouping.children
+    @background = @grouping.background.url
   end
 
   def update
-    lay_out = JSON.parse(params.require(:layout))
-    @grouping.children.each do |area|
-      x, y = lay_out[area.type][area.id.to_s].values_at('x', 'y')
-      if area.coordinate
-        area.coordinate.x = x
-        area.coordinate.y = y
-        render_exception(area.coordinate) unless area.coordinate.save
-      else
-        area.create_coordinate(x: x, y: y)
-        render_exception(area) unless area.save
+    if params[:grouping]
+      unless @grouping.update(grouping_params)
+        return render_exception(@grouping)
+      end
+    end
+    if params[:layout]
+      @grouping.children.each do |area|
+        x, y = coordinate(area)
+        if x.present? && y.present?
```

Comment:
Do you need the `present?`? Otherwise you can just do `if x && y`

Reply:
No I think you're right.